.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,107 @@
+name: benchmark
+
+on: workflow_call # TODO: add pull_request-labeled with "benchmark" label when specifying a label is supported
+
+jobs:
+  start-runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          # Ubuntu 22.04 128GB Storage AMI
+          ec2-image-id: ami-0ba430d4b7b64de57
+          ec2-instance-type: r6i.large
```

Comment:
Maybe this should be outside of the workflow in a secret, purely so that you can change sizes without a commit?

Reply:
prefer to have it visible for better debug and visibility